pkg/metrics/pushclient/pushclient_test.go

```diff
@@ -0,0 +1,92 @@
+package pushclient
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+// testClient is used as a mock client to test the functionality.
+type testClient struct {
+	// embed a stdlib http.Client
+	t              *testing.T
+	expectedString string
+	*http.Client
```

Comment:
https://golang.org/pkg/net/http/httptest/

see if that helps make this testing easier.

Reply:
Tried it but it made things complicated. Maybe I was not looking at it a little deeper. I'll check on that.